### PR TITLE
Updated contract filters

### PR DESCRIPTION
**Pull request by herpaderpaldent**
some improvements to contract filters 

requires: https://github.com/eveseat/services/pull/86

### CODE REVIEW

File: src/Http/Controllers/Corporation/ContractsController.php [hunk 1]
```diff
@@ -84,6 +86,54 @@ public function getContractsData(int $corporation_id)
...
+                        return $resolved_id->entity_id;
+                    });
+
+                $character_info_ids = CharacterInfo::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($character_info) {
+                        return $character_info->character_id;
+                    });
+
+                $query->whereIn('a.issuer_id', array_merge($resolved_ids->toArray(), $character_info_ids->toArray()));
```
Comment (warlof) [inline src/Http/Controllers/Corporation/ContractsController.php]: can't we use collection merge directly instead splitting the two upper and making a final array merge based on array ?

File: src/Http/Controllers/Corporation/ContractsController.php [hunk 2]
```diff
@@ -84,6 +86,54 @@ public function getContractsData(int $corporation_id)
...
+                        return $resolved_id->entity_id;
+                    });
+
+                $character_info_ids = CharacterInfo::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($character_info) {
+                        return $character_info->character_id;
+                    });
+
+                $query->whereIn('a.assignee_id', array_merge($resolved_ids->toArray(), $character_info_ids->toArray()));
```
Comment (warlof) [inline src/Http/Controllers/Corporation/ContractsController.php]: can't we use collection merge directly instead splitting the two upper and making a final array merge based on array ?

File: src/Http/Controllers/Corporation/ContractsController.php [hunk 3]
```diff
@@ -84,6 +86,54 @@ public function getContractsData(int $corporation_id)
...
+                        return $resolved_id->entity_id;
+                    });
+
+                $character_info_ids = CharacterInfo::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($character_info) {
+                        return $character_info->character_id;
+                    });
+
+                $query->whereIn('a.acceptor_id', array_merge($resolved_ids->toArray(), $character_info_ids->toArray()));
```
Comment (warlof) [inline src/Http/Controllers/Corporation/ContractsController.php]: can't we use collection merge directly instead splitting the two upper and making a final array merge based on array ?

**Comment by warlof** [comment]
Also, can you split the enhancement part from the hotfix change ?

**Comment by herpaderpaldent** [comment]
> Also, can you split the enhancement part from the hotfix change ?

yes i created a new PR for this: https://github.com/eveseat/web/pull/279

**Comment by herpaderpaldent** [comment]
Ok i introduced an service repository to reuse with any DataTablesFilter (which we have a few now)  https://github.com/eveseat/services/pull/86

File: src/Http/Controllers/Character/ContractsController.php [hunk 4]
```diff
@@ -174,46 +174,22 @@ public function getContractsData(int $character_id)
...
-                    ->get()
-                    ->map(function ($character_info) {
-                        return $character_info->character_id;
-                    });
-
-                $query->whereIn('a.assignee_id', array_merge($resolved_ids->toArray(), $character_info_ids->toArray()));
+
+                $resolved_ids = $this->getIdsForNames($keyword);
+
+                $query->whereIn('a.assignee_id', $resolved_ids->toArray());
```
Comment (leonjza) [inline src/Http/Controllers/Character/ContractsController.php]: Lets not alloc a new var here and instead just inline `$this->getIdsForNames($keyword)->toArray();`. Same would apply for the rest of the usage of this trait method going down.

**Comment by herpaderpaldent** [comment]
Allright i adapted everything should be good to go

**Comment by herpaderpaldent** [comment]
I have no idea what the latest force push means for me ... Does it or is this still good?

**Comment by warlof** [comment]
> I have no idea what the latest force push means for me ... Does it or is this still good?

I've only merge 3.0.x branch and reword some commits. This PR is unchanged.

**Pull request merged**